plugins/bigquery/dbt/adapters/bigquery/connections.py

```diff
@@ -573,3 +589,14 @@ def _is_retryable(error):
             e['reason'] == 'rateLimitExceeded' for e in error.errors):
         return True
     return False
+
+
+_SANITIZE_LABEL_PATTERN = re.compile(r"[^a-z0-9_-]")
+
+
+def _sanitize_label(value: str, max_length: int = 63) -> str:
+    """Return a legal value for a BigQuery label."""
+    value = value.lower()
+    value = _SANITIZE_LABEL_PATTERN.sub("_", value)
+    value = value[: max_length - 1]
+    return value
```

Comment:
Anecdotally, this has the effect of appending an extra underscore at the end of my non-dict comment, e.g. `comment: whatever` in `dbt_project.yml` becomes `query_comment: whatever_` in BigQuery, and `comment: something else` becomes `query_comment: something_else_`. Is that intended?

Reply:
I'm not sure why this is happening--there may be something upstream of this code adding trailing whitespace, because I don't get that behavior testing the function directly. I tried adding a `strip()`. Does that help?

Reply:
Looks like the `strip()` fixed it!

<img width="449" alt="Screen Shot 2021-03-09 at 10 39 33 AM" src="https://user-images.githubusercontent.com/13897643/110450706-bf379100-80c3-11eb-825c-5223009ab665.png">

(The numerous `_` in the top example are all special characters.)